runtime/drivers/azure/azure.go

```diff
@@ -0,0 +1,310 @@
+}
+
+// AsTransporter implements drivers.Connection.
+func (c *Connection) AsTransporter(from, to drivers.Handle) (drivers.Transporter, bool) {
+	return nil, false
+}
+
+func (c *Connection) AsFileStore() (drivers.FileStore, bool) {
+	return nil, false
+}
+
+// AsSQLStore implements drivers.Connection.
+func (c *Connection) AsSQLStore() (drivers.SQLStore, bool) {
+	return nil, false
+}
+
+// DownloadFiles returns a file iterator over objects stored in azure blob storage.
+func (c *Connection) DownloadFiles(ctx context.Context, source *drivers.BucketSource) (drivers.FileIterator, error) {
+	conf, err := parseSourceProperties(source.Properties)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse config: %w", err)
+	}
+
+	client, err := c.getClient(ctx, conf)
+	if err != nil {
+		return nil, err
+	}
+
+	// Create a *blob.Bucket.
+	bucketObj, err := azureblob.OpenBucket(ctx, client, nil)
+	if err != nil {
+		return nil, err
+	}
+	defer bucketObj.Close()
+
+	// prepare fetch configs
+	opts := rillblob.Options{
+		GlobMaxTotalSize:      conf.GlobMaxTotalSize,
+		GlobMaxObjectsMatched: conf.GlobMaxObjectsMatched,
+		GlobMaxObjectsListed:  conf.GlobMaxObjectsListed,
+		GlobPageSize:          conf.GlobPageSize,
+		GlobPattern:           conf.url.Path,
+		ExtractPolicy:         source.ExtractPolicy,
+	}
+
+	iter, err := rillblob.NewIterator(ctx, bucketObj, opts, c.logger)
+	if err != nil {
+		return nil, err
+	}
+
+	return iter, nil
+}
+
+type sourceProperties struct {
+	Path                  string `key:"path"`
+	GlobMaxTotalSize      int64  `mapstructure:"glob.max_total_size"`
+	GlobMaxObjectsMatched int    `mapstructure:"glob.max_objects_matched"`
+	GlobMaxObjectsListed  int64  `mapstructure:"glob.max_objects_listed"`
+	GlobPageSize          int    `mapstructure:"glob.page_size"`
+	url                   *globutil.URL
+}
+
+func parseSourceProperties(props map[string]any) (*sourceProperties, error) {
+	conf := &sourceProperties{}
+	err := mapstructure.Decode(props, conf)
+	if err != nil {
+		return nil, err
+	}
+	if !doublestar.ValidatePattern(conf.Path) {
+		return nil, fmt.Errorf("glob pattern %s is invalid", conf.Path)
+	}
+	url, err := globutil.ParseBucketURL(conf.Path)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse path %q, %w", conf.Path, err)
+	}
+	if url.Scheme != "azblob" {
+		return nil, fmt.Errorf("invalid scheme %q in path %q", url.Scheme, conf.Path)
+	}
+
+	conf.url = url
+	return conf, nil
+}
+
+// getClient returns a new azure blob client.
+func (c *Connection) getClient(ctx context.Context, conf *sourceProperties) (*container.Client, error) {
+	name := c.config.Account
+	key := c.config.Key
+
+	if c.config.AllowHostAccess {
+		name = os.Getenv("AZURE_STORAGE_ACCOUNT")
+		key = os.Getenv("AZURE_STORAGE_KEY")
+	}
+
+	credential, err := azblob.NewSharedKeyCredential(name, key)
+	if err != nil {
+		return nil, err
+	}
+
+	containerURL := fmt.Sprintf("https://%s.blob.core.windows.net/%s", name, conf.url.Host)
```

Comment:
If else can be used instead of creating anonymous client by default.
```
var client *container.Client
if key == "" {
    client, err = container.NewClientWithNoCredential(containerURL, nil)
} else {
    client, err = container.NewClientWithSharedKeyCredential(containerURL, credential, nil)
}
if err != nil {
    return nil, err
}
```